### PR TITLE
Soc 1657

### PR DESCRIPTION
**Pull request by justnpT**
Please do not merge after LGTM
after LGTM test groups should be updated in job:
http://jenkins:8080/view/Social/job/Test-Case-Automation-Login-Signup/

New groups for that job:
Login_anonCanLogInViaAuthModalModalWhenAddingFile
Login_anonCanLogInViaForcedLoginModalWhenAddingVide
Login_anonCanLoginViaAuthModalWhenAddingVideo
Login_anonCanLoginViaUserLoginPage
Login_anonCanLogInOnSpecialWatchListPage
Login_anonCanLogInViaForcedLoginModalWhenAddingPhoto
Login_anonCanLogInViaAuthModalWhenAddingPhoto

ForgottenPassword_anonCanRemindPasswordFromLoginDropdown
ForgottenPassword_anonCanRemindPasswordFromAuthModal
ForgottenPassword_anonCanRemindPasswordOnUserLoginSpecialPage
ForgottenPassword_anonCanRemindPasswordOnUserLoginSpecialPageUsingLowerCaseUserName
ForgottenPassword_anonCanRemindPasswordWhileCreatingWiki

Login_anonCanLoginOnUserLoginSpecialPage
Login_anonCanLoginFromDropdown
Login_anonCanLoginOnAuthModalFromGlobalNavigation
Login_anonCanLoginAsStaffOnUserLoginSpecialPage
Login_anonCanLoginAsStaffFromDropdown
Login_anonCanLoginAsStaffOnAuthModalFromGlobalNavigation
Login_anonCanLoginAsJapaneseUserOnUserLoginSpecialPage
Login_userWithoutAValidTokenGetsLoggedOut

Signin_anonCanNotSignInWithoutFillingCaptcha
Signin_anonCanNotSignInIfSheIsYoungerThanTwelve
Signin_anonCanNotSignInIfTheUsernameAlreadyExists
Signin_anonCanSignIn
Signin_anonCanSignInWhenCreatingNewWiki
Signin_userCanLoginWithoutConfirmingVerificationEmail
Signin_anonCanSignInWithUsernameContainingJapaneseSpecialCharacters

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
   Credentials credentials = Configuration.getCredentials();
 
-  @Test(groups = "ForcedLogin_001_newFile")
+  @Test(groups = "ForcedLogin_anonCanLogInViaForcedLoginModalWhenAddingFile")
```
Comment (kvas-damian) [inline src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java]: it's force login pop-up ;)

File: src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public void anonCanLogInViaForcedLoginModalWhenAddingFile() {
     specialPage.verifyUserLoggedIn(credentials.userName10);
   }
 
-  @Test(groups = "ForcedLogin_001_newFile")
+  @Test(groups = "Login_anonCanLogInViaAuthModalModalWhenAddingFile")
```
Comment (kvas-damian) [inline src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java]: `AuthModalModal`?

Comment (kvas-damian) [inline src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java]: Is it for purpose that group prefix doesn't match first test.

File: src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java [hunk 3]
```diff
@@ -98,8 +98,8 @@ public void anonCanLoginViaUserLoginPage() {
     Assertion.assertTrue(special.isStringInURL(URLsContent.SPECIAL_UPLOAD));
   }
 
-  @Test(groups = "ForcedLogin_004_notLoggedIn")
-  public void anonCanLoginOnSpecialWatchListPage() {
+  @Test(groups = "Login_anonCanLogInOnSpecialWatchListPage")
+  public void anonCanLogInOnSpecialWatchListPage() {
```
Comment (kvas-damian) [inline src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java]: if we decide to use `LogIn` which seems to be right we should be consistent with all method names ;)

File: src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java (hunk 2 above)
Comment (kvas-damian) [inline src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java]: And it's force login pop-up

File: src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java [hunk 4]
```diff
@@ -30,8 +30,8 @@
 public class SignUpTests extends NewTestTemplate {
   Credentials credentials = Configuration.getCredentials();
 
-  @Test(groups = {"SignUp_001", "SignUp"})
-  public void anonCanNotSignupWithoutFillingCaptcha() {
+  @Test(groups = {"Signin_anonCanNotSignInWithoutFillingCaptcha", "SignUp"})
```
Comment (kvas-damian) [inline src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java]: `Signin_...`
SignIn == Login
SignUp == Register

I would stick to `SignUp`.

**Comment by kvas-damian** [comment]
Adding groups that have almost the same name as method seems to be great candidate for automating. What do you think about it?

File: src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java (hunk 3 above)
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java]: Login before the underscore is a noun. It's the part of prefix that descrives the tested functionality.
Log in (LogIn) as two separate words is a verb and belongs to a sentence that describes the test case.

Comment (kvas-damian) [inline src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java]: That right:
`anonCanLogIn` here vs `anonCanLogin` in line 90.

File: src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java [hunk 5]
```diff
@@ -56,8 +56,8 @@ public void anonCanLogInViaAuthModalModalWhenAddingFile() {
     specialPage.verifyUserLoggedIn(credentials.userName10);
   }
 
-  @Test(groups = {"ForcedLogin_002_video", "Media"})
-  public void anonCanLoginViaForcedLoginModalWhenAddingVideo() {
+  @Test(groups = {"ForcedLogin_anonCanLogInViaForcedLoginPopupWhenAddingVideo", "Media"})
+  public void anonCanLoginViaForcedLoginPopupWhenAddingVideo() {
```
Comment (kvas-damian) [inline src/test/java/com/wikia/webdriver/testcases/logintests/ForcedLoginTests.java]: `Login` case incorrect ;)
In at least couple methods.

**Comment by justnpT** [comment]
@kvas-damian I agree it's a good place for automation. The two blockers for doing this are:
1. The rule of how we name groups is still evaluated: https://wikia-inc.atlassian.net/wiki/display/AUT/Group+names
2. Right now I don't have the necessary knowledge on what is the best way of automating this process.

**Comment by kvas-damian** [comment]
1. You can create follow up ticket when discussion ends.
2. You can ask other qm/dev for help ;)

**Comment by justnpT** [comment]
Here is the ticket: https://wikia-inc.atlassian.net/browse/SOC-2255

**Pull request merged**